plugin-flex-ts-template-v2/src/feature-library/dispositions/custom-components/DispositionTab/DispositionTab.tsx

```diff
@@ -19,101 +30,235 @@ export interface OwnProps {
   task?: ITask;
 }
 
-const DispositionTab = (props: OwnProps) => {
+interface StringPropsObject {
+  [key: string]: string;
+}
+interface BooleanPropsObject {
+  [key: string]: boolean;
+}
+
+const DispositionTab = ({ task }: OwnProps) => {
   const NOTES_MAX_LENGTH = 100;
   const [disposition, setDisposition] = useState('');
   const [notes, setNotes] = useState('');
+  const [customAttributes, setCustomAttributes] = useState({} as StringPropsObject);
+  const [groupOptions, setGroupOptions] = useState({} as BooleanPropsObject);
+  const [groupOptionsForQueue, setGroupOptionsForQueue] = useState({} as BooleanPropsObject);
+
+  const queueSid = task?.queueSid || '';
+  const queueName = task?.queueName || '';
+  const textAttributes = getTextAttributes(queueName);
+  const selectAttributes = getSelectAttributes(queueName);
+  const group = getMultiSelectGroup();
+  const groupForQueue = getQueueMultiSelectGroup(queueName);
+  const NO_OPTION_SELECTED = 'NoOptionSelected';
 
   const dispatch = useDispatch();
-  const { tasks } = useSelector((state: AppState) => state[reduxNamespace].dispositions as DispositionsState);
+  const { tasks: tasksFromRedux } = useSelector(
+    (state: AppState) => state[reduxNamespace].dispositions as DispositionsState,
+  );
 
   const updateStore = () => {
-    if (!props.task) return;
-
-    let payload = {
-      taskSid: props.task.taskSid,
-      disposition: '',
-      notes: '',
+    if (!task) return;
+    const payload = {
+      taskSid: task?.taskSid,
+      disposition: disposition ? disposition : '',
+      notes: notes ? notes : '',
+      custom_attributes: { ...customAttributes },
     };
-
-    if (tasks[props.task.taskSid]) {
-      payload = {
-        ...payload,
-        ...tasks[props.task.taskSid],
-      };
-    }
-
-    if (disposition) {
-      payload.disposition = disposition;
-    }
-
-    if (isNotesEnabled() && notes) {
-      payload.notes = notes;
-    }
-
     dispatch(updateDisposition(payload));
   };
 
   const updateStoreDebounced = debounce(updateStore, 250, { maxWait: 1000 });
```

Comment:
I faced a similar problem in a different code base and I realized the `debounce` here is not used correctly. You are not debouncing anything here. You are just delaying the execution of `updateStore` for `250ms`. So if you type 10 characters in that time, at the end, you will send 10 requests to the `updateStore` function.